src/services/bookings.ts

```diff
@@ -20,6 +22,39 @@ export async function getUserBookings(
   })
 }
 
+async function hasExistingConflicts(startTime: Date, endTime: Date) {
+  const conflicting = await prisma.booking.findMany({
+    where: {
+      start: {
+        lt: endTime,
+      },
+    },
+    orderBy: {
+      end: 'desc',
+    },
+  })
+
+  return conflicting.length > 0 && conflicting[0].end > startTime
+}
+
+async function isStackedBooking(orgId: number, startTime: Date) {
+  const lastOrgBooking = await prisma.booking.findFirst({
+    where: {
+      orgId: orgId,
+    },
+    orderBy: {
+      end: 'desc',
+    },
+  })
+
+  if (!lastOrgBooking) {
+    return true
+  }
+
+  return startTime.getTime() - lastOrgBooking.end.getTime() * 60 * 1000 < MIN_SLOTS_BETWEEN_BOOKINGS * DURATION_PER_SLOT
```

Comment:
hmm, what about this case:

existing: 6 pm - 8 pm 5/1/23, 10am - 12pm 5/1/23
new: 12pm - 2pm 5/1/23

The new booking has startTime  - lastOrgBooking (6pm-8pm) >MIN_SLOTS_BETWEEN_BOOKINGS * DURATION_PER_SLOT. But it is also a stacked booking since it follows right after the booking that ended at 12pm?

Reply:
Thanks for the catch! Yup it is wrong haha I will use ur functions to be safer since I think you have written unit tests for them

Reply:
Wanted to ask on your opinion for checkConflictingBooking as well, do you see any issues with the logic I have written? It might possibly a more concise way of implementing the logic, but am not sure if I missed anything out.

Reply:
for `hasExistingConflicts`, it doesn't really check for overlapping bookings like:
A:           |               |
B:                |                  |

OR

Reply:
yep `hasExistingConflict` lgtm!

I think it is much neater than the one i wrote !